### DISCUSSION TITLE
Order of installation

### DISCUSSION THREAD

**Discussion by jeanpat** [PyTorchLightning/lightning-flash#1471] [Q&A]
Hello,
I try to reinstall lightning-flash in clean virtual environment. If I start with:
`
pip install 'lightning-flash[image]'`
The install starts to install the cpu version of pytorch 1.12:
![image](https://user-images.githubusercontent.com/693574/196632245-bc3b996e-0d26-4c70-9d9e-6b56c58cc24d.png)

Should I start to install the latest gpu version followed by the install of lightning-flash from source from source?
```
conda install pytorch torchvision torchaudio cudatoolkit=11.6 -c pytorch -c conda-forge
pip install 'git+https://github.com/PyTorchLightning/lightning-flash.git#egg=lightning-flash[image]'
```
Thanks

**Comment by ethanwharris** [comment]
Hey @jeanpat yes, I think the right way would be to install pytorch manually first and then install Flash (either from source or pip should both be fine). That should prevent the Flash install from overriding the pytorch version.

Hope that helps 😃

**Reply by jeanpat** [reply]
Thanks Ethan,
I'll try that, which version of cuda is recommended?

**Reply by ethanwharris** [reply]
I think all should be fine, but it depends which torch version as I think they just match each torch version to a couple of cuda versions
